test/parallel_api/iterator/transform_output_iterator.pass.cpp

```diff
@@ -0,0 +1,153 @@
+// -*- C++ -*-
+//===-- transform_output_iterator.pass.cpp ---------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include _PSTL_TEST_HEADER(execution)
+#include _PSTL_TEST_HEADER(algorithm)
+#include _PSTL_TEST_HEADER(iterator)
+
+#include "support/utils.h"
+
+#include <tuple>
+
+using namespace TestUtils;
+
+#if TEST_DPCPP_BACKEND_PRESENT
+
+DEFINE_TEST(test_copy)
+{
+    DEFINE_TEST_CONSTRUCTOR(test_copy)
+
+    template <typename ExecutionPolicy, typename Iterator1, typename Iterator2, typename Size, typename TExpectedValue>
+    void operator()(ExecutionPolicy&& exec, Iterator1 first1, Iterator1 last1, Iterator2 first2, Size n, TExpectedValue expected_value)
+    {
+        TestDataTransfer<UDTKind::eVals, Size> host_vals(*this, n);
+
+        ::std::copy(::std::forward<ExecutionPolicy>(exec), first1, last1, first2);
+
+        host_vals.retrieve_data();
+        auto res_begin = host_vals.get();
+        for(int i = 0; i != n; ++i) {
```

Comment:
Is it possible to use our macro EXPECT_EQ_N instead of an explicit loop?

Reply:
Yes, I will use this macro for the test.